semantic/graph.go

```diff
@@ -137,13 +141,55 @@ func (p *Program) Copy() Node {
 	return np
 }
 
+type PackageClause struct {
+	loc `json:"-"`
+
+	Name *Identifier `json:"name"`
+}
+
+func (*PackageClause) NodeType() string { return "PackageClause" }
+
+func (p *PackageClause) Copy() Node {
+	if p == nil {
+		return p
+	}
+	np := new(PackageClause)
+	*np = *p
+
+	np.Name = p.Name.Copy().(*Identifier)
+
+	return np
+}
+
+type ImportDeclaration struct {
+	loc `json:"-"`
+
+	As   *Identifier    `json:"as"`
+	Path *StringLiteral `json:"path"`
+}
+
+func (*ImportDeclaration) NodeType() string { return "ImportDeclaration" }
+
+func (d *ImportDeclaration) Copy() Node {
+	if d == nil {
+		return d
+	}
+	nd := new(ImportDeclaration)
+	*nd = *d
+
+	nd.As = d.As.Copy().(*Identifier)
```

Comment:
It is safe to call Copy when `As` is nil.

The implementation of `Identifier.Copy` explicitly handles this case:

```
func (i *Identifier) Copy() Node {
	if i == nil {
		return i
	}
	ni := new(Identifier)
	*ni = *i

	return ni
}
```

In fact all of the Copy methods do. Also note that `i` is returned instead of a bare `nil`. This is because `nils` in Go are typed and `i`'s type is `*Identifier` and a bare `nil` would have type `Node`, but since we type assert the result as an `*Identifier` it would panic if we  returned a `nil` with type `Node`.